src/reducers/notifications/types.ts

```diff
@@ -0,0 +1,93 @@
+import { AppError } from '../../lib/errors'
+import { NotificationHandler } from '../../ui/notifications/types'
+
+export enum NotificationType {
+  error = 'error',
+  info = 'info',
+  warning = 'warning',
+}
+
+/**
+ * Depending on the notification severity, it might be rendered differently on the UI. For example:
+ * low - simple top notification banner
+ * medium - top notification banner with accent colors on title and action button
+ * high - full screen overlay with call to action
+ */
+
+export enum NotificationSeverity {
+  low = 'low',
+  medium = 'medium',
+  high = 'high',
+}
```

Comment:
I think the naming can be improved here. If this severity values are a direct indicator how the notification is displayed, this should than also be reflected in it's values, like e.g. `topBanner`, `actionTopBanner` and `fullScreen`. Is the severity somehow related to the NotificationType?

Reply:
We might have different notification handlers / consumers. The intent was to decouple the error structure from the UI as much as possible, i.e. not make assumptions about the way they are rendered.

With the examples you shared, the orange border notification (i.e. when a scan failed) would need different keys, or wouldn't idiomatically fit in (currently represented as a local error notification of low severity).

Currently the notification container HOC is responsible for rendering errors or notifications of different severity how it sees fit.

Reply:
Related relationships between type and severity, depends on the handler, but generally yes. Given the top bar notification handler:

| type | severity | rendered as |
|------|----------|-------------
| info | low         |    top bar, normal colors         | 
| info |medium |   top bar, accent colors          | 
| info |        high  |  full screen overlay           |
| warning | low         |   top bar normal colors          | 
| warning |medium |    top bar accent colors         |  
| warning |        high  | full screen confirmation screen (i.e. are you sure you want to reject an issued credential)      |
| error | low         |  top bar, normal colors           |
| error |medium |    top bar, accent colors         |
| error |        high  |        full screen overlay, with the report flow in     |

Reply:
But this matrix will look different depending on the notification handler of course

Reply:
I think one of the axis can be safely done away with, since there only seem to be really 2-3 ways things might look different, currently based on low/medium severity deciding whether or not to use accent colors. (Since all the "full screen" ones are not really notifications as we have discussed)

I would suggest we do away with "severity" though, and use colors based on type